frame/bridge/ethereum/issuing/src/lib.rs

```diff
@@ -0,0 +1,423 @@
+};
+use darwinia_relay_primitives::relay_authorities::*;
+use darwinia_support::{balance::lock::*, traits::EthereumReceipt};
+use dp_evm::CallOrCreateInfo;
+use dvm_ethereum::{TransactionAction, TransactionSignature};
+use ethereum_primitives::{
+	receipt::{EthereumTransactionIndex, LogEntry},
+	EthereumAddress,
+};
+use types::*;
+
+pub trait Config: dvm_ethereum::Config {
+	type Event: From<Event<Self>> + Into<<Self as frame_system::Config>::Event>;
+
+	type ModuleId: Get<ModuleId>;
+
+	type RingCurrency: LockableCurrency<Self::AccountId, Moment = Self::BlockNumber>;
+
+	type EthereumRelay: EthereumReceipt<Self::AccountId, RingBalance<Self>>;
+	type EcdsaAuthorities: RelayAuthorityProtocol<Self::BlockNumber, Signer = EthereumAddress>;
+
+	type WeightInfo: WeightInfo;
+}
+
+decl_error! {
+	/// Issuing pallet errors.
+	pub enum Error for Module<T: Config> {
+		/// Invalid Issuing System Account
+		InvalidIssuingAccount,
+		/// assert ar
+		AssetAR,
+		/// ReceiptProofInv
+		ReceiptProofInv,
+		/// LogEntryNE
+		LogEntryNE,
+		/// EthLogPF
+		EthLogPF,
+		/// StringCF
+		StringCF,
+		/// Unit
+		UintCF,
+		/// Address - CONVERSION FAILED
+		AddressCF,
+		/// encode erc20 tx failed
+		InvalidEncodeERC20,
+		/// encode mint tx failed
+		InvalidMintEncoding,
+		/// invalid ethereum address length
+		InvalidAddressLen,
+		/// decode input value error
+		InvalidInputData,
+	}
+}
+
+decl_event! {
+	pub enum Event<T>
+	where
+		AccountId = AccountId<T>,
+	{
+		/// register new erc20 token
+		RegisterErc20(AccountId, EthereumAddress),
+		/// redeem erc20 token
+		RedeemErc20(AccountId, EthereumAddress),
+		/// erc20 created
+		CreateErc20(EthereumAddress),
+		/// burn event
+		/// type: 1, backing, recipient, source, target, value
+		BurnToken(u8, EthereumAddress, EthereumAddress, EthereumAddress, EthereumAddress, U256),
+		/// token registered event
+		/// type: u8 = 0, backing, source(origin erc20), target(mapped erc20)
+		TokenRegistered(u8, EthereumAddress, EthereumAddress, EthereumAddress),
+	}
+}
+
+decl_storage! {
+	trait Store for Module<T: Config> as DarwiniaEthereumIssuing {
+		pub MappingFactoryAddress get(fn mapping_factory_address) config(): EthereumAddress;
+		pub EthereumBackingAddress get(fn ethereum_backing_address) config(): EthereumAddress;
+		pub VerifiedIssuingProof
+			get(fn verified_issuing_proof)
+			: map hasher(blake2_128_concat) EthereumTransactionIndex => bool = false;
+		pub BurnTokenEvents get(fn burn_token_events): Vec<<T as frame_system::Config>::Event>;
+	}
+}
+
+decl_module! {
+	pub struct Module<T: Config> for enum Call
+	where
+		origin: T::Origin
+	{
+		fn deposit_event() = default;
+
+		fn on_initialize(_n: BlockNumber<T>) -> Weight {
+			<BurnTokenEvents<T>>::kill();
+			0
+		}
+
+		#[weight = <T as darwinia_evm::Config>::GasWeightMapping::gas_to_weight(0x100000)]
+		pub fn register_erc20(origin, proof: EthereumReceiptProofThing<T>) {
+			log::info!(target: "darwinia-issuing", "start to register erc20 token");
```

Comment:
log will target the crate's name automactically

Reply:
Also you use info here. These log will always be printed.

Reply:
forgot to remove all these logs used to debug.